waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -83,42 +83,67 @@ genMigrationsDir spec =
 postWriteDbGeneratorActions :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO ([GeneratorWarning], [GeneratorError])
 postWriteDbGeneratorActions spec dstDir = do
-  dbGeneratorWarnings <- maybeToList <$> warnIfDbSchemaChangedSinceLastMigration spec dstDir
+  dbGeneratorWarnings <- maybeToList <$> warnIfDbNeedsMigration spec dstDir
   dbGeneratorErrors <- maybeToList <$> genPrismaClient spec dstDir
   return (dbGeneratorWarnings, dbGeneratorErrors)
 
--- | Checks if user needs to run `wasp db migrate-dev` due to changes they did in schema.prisma, and if so, returns a warning.
+-- | Checks if user needs to run `wasp db migrate-dev` due to changes they made in schema.prisma, and if so, returns a warning.
 -- When doing this, it looks at schema.prisma in the generated project.
 --
 -- This function makes following assumptions:
 --  - schema.prisma will exist in the generated project even if no Entities were defined.
 --    Due to how Prisma itself works, this assumption is currently fulfilled.
---  - schema.prisma.wasp-checksum contains the checksum of the schema.prisma as it was during the last `wasp db migrate-dev`.
+--  - schema.prisma.wasp-last-db-concurrence-checksum contains the checksum of the schema.prisma as it was when we last ensured it matched the DB.
 --
 -- Given that, there are two cases in which we wish to warn the user to run `wasp db migrate-dev`:
--- (1) If schema.prisma.wasp-checksum exists, but is not equal to checksum(schema.prisma), we know they made changes to schema.prisma and should migrate.
--- (2) If schema.prisma.wasp-checksum does not exist, but the user has entities defined in schema.prisma (and thus, AppSpec).
+-- (1) If schema.prisma.wasp-last-db-concurrence-checksum exists, but is not equal to checksum(schema.prisma), we know they made changes to schema.prisma and should migrate.
+-- (2) If schema.prisma.wasp-last-db-concurrence-checksum does not exist, but the user has entities defined in schema.prisma (and thus, AppSpec).
 --     This could imply they have never migrated locally, or that they have but are simply missing their generated project dir.
 --     Common scenarios for the second warning include:
---       - After a fresh checkout, or after `wasp clean`; possible false positives in these cases, but for safety, it's still preferable to warn.
+--       - After a fresh checkout, or after `wasp clean`.
 --       - When they previously had no entities and just added their first.
-warnIfDbSchemaChangedSinceLastMigration :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO (Maybe GeneratorWarning)
-warnIfDbSchemaChangedSinceLastMigration spec projectRootDir = do
+--     In either of those scenarios, validate against DB itself to avoid redundant warnings.
+--
+--     NOTE: As one final optimization, if they do not have a schema.prisma.wasp-last-db-concurrence-checksum but the schema is
+--     in sync with the databse, we generate that file to avoid future checks.
```

Comment:
Makes sense! What were we doing before, we were not generating that file but where letting them to run wasp db migrate-dev?

Reply:
Before, we only generated the file on migrate.

Reply:
Ok! Makes perfect sense to me that we generate it every time now, at the end that is its name now -> the last checksum that we verified.